static/app/components/profiling/ThreadSelector.tsx

```diff
@@ -0,0 +1,220 @@
+      }
+      props.onProfileIndexChange(index);
+    },
+    [props.onProfileIndexChange, props.profileGroup]
+  );
+
+  React.useEffect(() => {
+    const tagNamesToPreventStealingFocusFrom = new Set<Element['tagName']>([
+      'INPUT',
+      'TEXTAREA',
+    ]);
+    function handleKeyDown(evt) {
+      if (tagNamesToPreventStealingFocusFrom.has(evt.target.tagName)) {
+        return;
+      }
+      if (evt.key === 't' && !open) {
+        evt.preventDefault();
+        setOpen(true);
+      }
+      if (evt.key === 'Escape' && open) {
+        setOpen(false);
+      }
+    }
+
+    document.addEventListener('keydown', handleKeyDown);
+    return () => document.removeEventListener('keydown', handleKeyDown);
+  }, [open]);
+
+  useOnClickOutside(containerRef, () => {
+    setOpen(false);
+  });
+
+  return (
+    <div>
+      <Button size="zero" borderless onClick={() => setOpen(true)}>
+        {props.profileGroup.profiles[props.activeProfileIndex]?.name ?? 'Select Thread'}
+      </Button>
+      <AutoComplete
+        // @ts-ignore the type is typed as any
+        onSelect={handleSelectItem}
+        closeOnSelect={false}
+        isOpen={open}
+      >
+        {({getInputProps, getItemProps, isOpen, inputValue, highlightedIndex}) => {
+          const sortedProfiles = sortProfiles(props.profileGroup.profiles);
+          const filteredProfiles = inputValue
+            ? new Fuse(sortedProfiles, {includeMatches: true, keys: ['name']}).search(
+                inputValue
+              )
+            : sortedProfiles.map(p => ({item: p, matches: [], score: 1}));
+
+          return isOpen ? (
+            <ThreadSelectorContainer ref={containerRef}>
+              <Input
+                type="search"
+                autoFocus
+                {...getInputProps({
+                  onKeyDown: evt => {
+                    if (evt.key === 'Escape') {
+                      setOpen(false);
+                    }
+                  },
+                })}
+              />
+              <DropdownBox>
+                {filteredProfiles.length > 0 ? (
+                  filteredProfiles.map((profile, index) => {
+                    const activeProfile =
+                      props.profileGroup.profiles[props.activeProfileIndex];
+
+                    return (
+                      <SearchResult
+                        key={index}
+                        highlighted={index === highlightedIndex}
+                        ref={ref =>
+                          index === highlightedIndex
+                            ? ref?.scrollIntoView({block: 'nearest'})
+                            : null
+                        }
+                        {...getItemProps({
+                          // @ts-ignore the type is typed as any
+                          item: profile,
+                          index,
+                        })}
+                      >
+                        {activeProfile.name === profile.item.name ? (
```

Comment:
Any chance here that the active profile will be undefined?

Also, we're assuming the profile names are unique here, is that a safe assumption?

Reply:
No, we always have an active profile. iirc a valid trace should have least 1 all the time